### ISSUE TITLE
[Bug]: Canonical url error with Tag page

### ISSUE THREAD

**Issue by GamertechAU** [GetPublii/Publii#1844]
### Operating system

Fedora Linux 41

### Publii version

0.46.2 (build 16966)

### Post editor

None

### Bug description

When previewing/publishing my site with a tag page, the canonical url for the tag pages doesn't match with the actual url. It's adding the POSTS_PREFIX to the canonical url but not the actual.

Example:
Actual page url: https://example.com/TAGS_PREFIX/tag-slug
Canonical url: https://example.com/POSTS_PREFIX/TAGS_PREFIX/tag-slug

Paginated urls are the same, just with /page/2/ attached.

If I enable `Put tags prefix after posts prefix` then both the actual and canonical url add an additional POSTS_PREFIX
ie. Canonical: https://example.com/POSTS_PREFIX/POSTS_PREFIX/TAGS_PREFIX/tag-slug

Theme: Tattoo
Homepage is set to custom page
Use pretty URL's is enabled
Tag canonical box is empty
No plugins installed

No other page type seems to be affected including posts.

Expected outcome would be to have the canonical url match the actual.

### Steps to reproduce

I haven't found a way to fix it yet, so unsure how to reproduce it.

1- Create a tag in the Tags section of Publii
2- Preview changes to generate site
3- Browse to the Tag page
4- Inspect source code and search for the Canonical url
5- Observe extra POSTS_PREFIX is included

### Relevant log output

```shell
Only error shown was:

(node:95731) Warning: No such label 'BLOG PAGINATION' for console.timeEnd()

In rendering-deployment-errors.log and rendering-errors.log
```

**Comment by GamertechAU** [comment]
Just to add, the Publii-generated sitemap shows the url's without the POSTS_PREFIX. It shows the actual url used. So this is only affecting the canonical url of tag pages.

**Comment by GamertechAU** [comment]
Disabling Pretty URL's fixes the canonical issue, but isn't desired as then the root becomes a mess and loses the url hierarchy.

**Comment by dziudek** [comment]
@GamertechAU - this issue is related to https://github.com/GetPublii/Publii/discussions/1820 and will be fixed in upcoming v.0.46.3 - the post prefix has been added unnecessary two times (for some cases) and it also could affect canonical URLs

**Comment by GamertechAU** [comment]
Thanks @dziudek I did see that discussion but it mentioned the post page specifically while saying their canonical was correct. My issue is the reverse and tag-specific so believed it was a separate issue. I can confirm it also affects my og:url on the same page.

If it's known and fixed, then thanks for your work :)

**Comment by GamertechAU** [comment]
Can confirm, fixed in v0.46.3. Thanks :)
